install/0000_30_machine-api-operator_00_credentials-request.yaml

```diff
@@ -96,38 +96,27 @@ spec:
     - Microsoft.Compute/virtualMachines/read
     - Microsoft.Compute/virtualMachines/write
     - Microsoft.ManagedIdentity/userAssignedIdentities/assign/action
-    - Microsoft.Network/applicationSecurityGroups/delete
     - Microsoft.Network/applicationSecurityGroups/read
-    - Microsoft.Network/applicationSecurityGroups/write
     - Microsoft.Network/loadBalancers/backendAddressPools/join/action
-    - Microsoft.Network/loadBalancers/delete
     - Microsoft.Network/loadBalancers/read
     - Microsoft.Network/loadBalancers/write
     - Microsoft.Network/networkInterfaces/delete
     - Microsoft.Network/networkInterfaces/join/action
     - Microsoft.Network/networkInterfaces/loadBalancers/read
     - Microsoft.Network/networkInterfaces/read
     - Microsoft.Network/networkInterfaces/write
-    - Microsoft.Network/networkSecurityGroups/delete
     - Microsoft.Network/networkSecurityGroups/read
     - Microsoft.Network/networkSecurityGroups/write
     - Microsoft.Network/publicIPAddresses/delete
     - Microsoft.Network/publicIPAddresses/join/action
     - Microsoft.Network/publicIPAddresses/read
     - Microsoft.Network/publicIPAddresses/write
-    - Microsoft.Network/routeTables/delete
```

Comment:
A lot of these permissions were included because [code paths exist](https://github.com/openshift/machine-api-provider-azure/blob/458e10c9b4f57f1c26613da7e0c2b5feb7118921/pkg/cloud/azure/services/routetables/routetables.go#L86) that use them in the azure provider

Reply:
ACK, will depend on if we utilize the code in MAO. May require some extra scrutiny.

Reply:
This also smells like the sort of thing we should do some pre-merge testing on but it would be great to reduce the permission set to only those that are necessary.

Reply:
Agree, can we check that these Azure SDK wrapper functions that are implemented, aren't being called before we remove the permissions? 

@sunzhaohua2 Who's the appropriate person to do some pre-merge testing on this PR?

Reply:
@JoelSpeed will this only affect our team's cases, or we also need to check network team's cases

Reply:
Should only affect the Machine API provider Azure related cases, not expecting any impact on CCM or networking in general

Reply:
Thanks, I am running some regression testing,  will update result in bug.